src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -68,45 +68,18 @@ namespace po = boost::program_options;
 // ------ L O C A L   F U N C T I O N S ---------------------------------------
 
 namespace {
-/*
- * Update shell on the board for auto flash
- */
-static void 
-update_shell(unsigned int index, const std::string& primary, const std::string& secondary)
-{
-  Flasher flasher(index);
-  if(!flasher.isValid())
-    throw xrt_core::error(boost::str(boost::format("%d is an invalid index") % index));
-
-  if (primary.empty())
-    throw xrt_core::error("Shell not specified");
-
-  auto pri = std::make_unique<firmwareImage>(primary.c_str(), MCS_FIRMWARE_PRIMARY);
-  if (pri->fail())
-    throw xrt_core::error(boost::str(boost::format("Failed to read %s") % primary));
-
-  std::unique_ptr<firmwareImage> sec;
-  if (!secondary.empty()) {
-    sec = std::make_unique<firmwareImage>(secondary.c_str(),
-      MCS_FIRMWARE_SECONDARY);
-    if (sec->fail())
-      sec = nullptr;
-  }
-  
-  if (flasher.upgradeFirmware("", pri.get(), sec.get()) != 0)
-    throw xrt_core::error("Failed to update base");
-  
-  std::cout << boost::format("%-8s : %s \n") % "INFO" % "Base flash image has been programmed successfully.";
-}
 
-/*
- * Update shell on the board for manual flash
+/**
+ * @brief Update shell on the board for manual flash
+ * 
+ * @param[in] index The index of the board to be flashed
+ * @param[in] image_paths The images to plash onto the board
```

Comment:
plash?

Reply:
We can have several images? What is the semantics? Loading all of them on the board?

Reply:
There was once a board that took two images when being flashed as far as I was  told. So it is in fact a vector of images to flash. It was much cleaner than having multiple image variables.